view/frontend/templates/js/replacejs.phtml

```diff
@@ -1695,9 +1695,37 @@ if (!$block->isSaveCartInSections()) { ?>
         /////////////////////////////////////////////////////
         var createRequest = false;
         var allowAutoOpen = true;
-        var wasConfigureCalled = false;
+        var waitingForResolvingPromises = false;
         var oldBoltCartValue = "";
         var BC;
+        var hintBarrier;
+        var cartBarrier;
+        var countConfigureCalls = 0;
+
+        var callConfigureWithPromises = function() {
+            cartBarrier = boltBarrier();
+            hintBarrier = boltBarrier();
+            BC = BoltCheckout.configure(cartBarrier.promise, hintBarrier.promise, callbacks);
+            waitingForResolvingPromises = true;
+            countConfigureCalls++;
+            var localCallNumber = countConfigureCalls;
+            // invalidate bolt cart if we didn't resolve promises for some reasons
+            setTimeout(function () {
+                if (waitingForResolvingPromises && (localCallNumber === countConfigureCalls)) {
```

Comment:
Not understanding the exact purpose of the second comparison here. @vitaliyreznikov can you explain?

Reply:
If, for example we call this function twice with an interval of 3 seconds we don't want the first call invalidates cart because the second call still works and we are waiting for the result.